pkg/transport/keepalive_listener.go

```diff
@@ -23,15 +23,10 @@ import (
 
 // NewKeepAliveListener returns a listener that listens on the given address.
 // http://tldp.org/HOWTO/TCP-Keepalive-HOWTO/overview.html
-func NewKeepAliveListener(addr string, scheme string, info TLSInfo) (net.Listener, error) {
-	l, err := net.Listen("tcp", addr)
-	if err != nil {
-		return nil, err
-	}
-
+func NewKeepAliveListener(l net.Listener, scheme string, info TLSInfo) (net.Listener, error) {
```

Comment:
@heyitsanthony Right. The TLS one must be the most outside listener. So we need to prepare inner listener and then pass in that listener. I will document it.